File: templates/Enterprise-Template/src/csharp/EnterpriseBotTemplate/Bot Framework/Dialogs/SignIn/SignInResponses.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using $safeprojectname$.Dialogs.SignIn.Resources;
using $safeprojectname$.Extensions;
using Microsoft.Bot.Builder.TemplateManager;

namespace $safeprojectname$.Dialogs.Shared
{
    public class SignInResponses : TemplateManagerWithVoice
    {
        // Constants
        public const string SignInPrompt = "namePrompt";
        public const string Succeeded = "haveName";
        public const string Failed = "emailPrompt";

        // Fields
        private static LanguageTemplateDictionary _responseTemplates = new LanguageTemplateDictionary
        {
            ["default"] = new TemplateIdMap
            {
                {
                    SignInPrompt,
                    (context, data) => SignInStrings.PROMPT
                },
                {
                    Succeeded,
                    (context, data) => string.Format(SignInStrings.SUCCEEDED, data.name)
                },
                {
                    Failed,
                    (context, data) => SignInStrings.FAILED
                },
            },
            ["en"] = new TemplateIdMap { },
            ["fr"] = new TemplateIdMap { },
        };

        public SignInResponses()
        {
            this.Register(new DictionaryRenderer(_responseTemplates));
        }
    }
}

File: templates/Enterprise-Template/src/typescript/src/dialogs/signin/signInResponses.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License

import { TurnContext } from "botbuilder";
import { DictionaryRenderer, LanguageTemplateDictionary, TemplateFunction } from "../templateManager/dictionaryRenderer";
import { TemplateManager } from "../templateManager/templateManager";
import * as i18n from "i18n";

export class SignInResponses extends TemplateManager {
    // Constants
    public static readonly SignInPrompt: string = "namePrompt";
    public static readonly Succeeded: string = "haveName";
    public static readonly Failed: string = "emailPrompt";

    // Fields
    private static readonly _responseTemplates: LanguageTemplateDictionary = new Map([
        ["default", new Map([
            [SignInResponses.SignInPrompt, SignInResponses.fromResources("signIn.prompt")],
            [SignInResponses.Failed, SignInResponses.fromResources("signIn.failed")],
            [SignInResponses.Succeeded, async (context: TurnContext, data: any) => {
                const value = i18n.__("signIn.succeeded");
                return value.replace("{0}", data.name);
            }],
        ])]
    ]);

    private static fromResources(name: string): TemplateFunction {
        return (context: TurnContext, data: any) => Promise.resolve(i18n.__(name));
    }

    constructor() {
        super();
        this.register(new DictionaryRenderer(SignInResponses._responseTemplates));
    }
}
